src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -210,6 +216,24 @@ public KeyRangeIterator.Builder getIndexQueryResults(Collection<Expression> expr
         return builder;
     }
 
+    private void maybeTriggerGuardrails(QueryViewBuilder.QueryView queryView)
+    {
+        if (!cfs.isLeveledCompaction())
+            return;
+
+        int referencedIndexes = queryView.referencedIndexes.size();
+
+        if (Guardrails.maximumReferencedIndexesWithNonParitionRestrictedQuery.failsOn(referencedIndexes, null))
```

Comment:
here I am just checking if it is going to fail, we do no need any ClientState for that.

Reply:
We should follow the pattern of `LocalReadSizeTooLargeException` and trace here, using a common error message for the exception, etc.